hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt

```diff
@@ -63,15 +65,15 @@ class HmppsQueueFactory(
   fun createSqsDlqClient(queueId: String, queueConfig: QueueConfig, hmppsSqsProperties: HmppsSqsProperties): AmazonSQS =
     with(hmppsSqsProperties) {
       when (provider) {
-        "aws" -> amazonSqsFactory.awsSqsDlqClient(queueId, queueConfig.dlqName, queueConfig.dlqAccessKeyId, queueConfig.dlqSecretAccessKey, region, queueConfig.asyncDlqClient)
+        "aws" -> return amazonSqsFactory.awsSqsDlqClient(queueId, queueConfig.dlqName!!, queueConfig.dlqAccessKeyId, queueConfig.dlqSecretAccessKey, region, queueConfig.asyncDlqClient)
```

Comment:
Rather than the bangbang here it might be better to put a defensive clause in - i.e. just throw an exception if dlqName is null (then  you won't need the bang bang). Because truth be told the dlqName _could_ be null if someone calls it directly for example - in which case it's better to have a controlled exception with some nice text instead of some random null pointer.

Reply:
Added